src/dom.js

```diff
@@ -375,6 +375,21 @@ export function childElementByTag(parent, tagName) {
 
 
 /**
+ * A wrapper around native querySelectorAll to use :scope when possible.
+ * @param {!Element} parent
+ * @param {string} selector
+ * @returns {!Array.<!Element>}
+ */
+export function querySelectorAll(parent, selector) {
+  if (scopeSelectorSupported) {
+    const scopedSelectors = selector.split(',').map(sel => `:scope ${sel}`);
+    return toArray(parent.querySelectorAll(scopedSelectors.join(',')));
```

Comment:
I guess I was trying to stick to the same return signature to other methods in `dom.js`, but I think for this one, we can go with the `NodeList` return instead since that's always what is to be returned. Will change.